streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -83,6 +99,8 @@ public void process(final K key, final V value) {
             throw new StreamsException("Invalid (negative) timestamp of " + timestamp + " for output record <" + key + ":" + value + ">.");
         }
 
+        final String topic = topicExtractor != null ? topicExtractor.extract(key, value, this.context.recordContext()) : this.topic;
```

Comment:
I am wondering, if we should always use a `TopicNameExtractor`. Ie, if a topic is given, we create a `TopicNameExtractor` that always return the topic name?

Reply:
I tend to not do that because it would be one extra function call on each record to send to.

Reply:
Not sure what is more performant: a function call (that might get inlined via JIT) or evaluating the branch-condition that would also be optimized during runtime quite a bit.... Was just a thought. Not sure if there will be a big difference between both during runtime.

Reply:
I'll go ahead and make the change, with some optimizations on runtime window partitioner check.